streams/test-utils/src/test/java/org/apache/kafka/streams/test/wordcount/WindowedWordCountProcessorTest.java

```diff
@@ -136,54 +133,18 @@ public void shouldWorkWithPersistentStore() throws IOException {
             context.scheduledPunctuators().get(0).getPunctuator().punctuate(1_000L);
 
             // finally, we can verify the output.
-            final Iterator<MockProcessorContext.CapturedForward> capturedForwards = context.forwarded().iterator();
-            assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[alpha@100/200]", "2")));
-            assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[beta@100/200]", "1")));
-            assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[delta@200/300]", "1")));
-            assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[gamma@100/200]", "1")));
-            assertThat(capturedForwards.next().keyValue(), is(new KeyValue<>("[gamma@200/300]", "1")));
-            assertThat(capturedForwards.hasNext(), is(false));
+            final List<CapturedForward<? extends String, ? extends String>> capturedForwards = context.forwarded();
+            final List<CapturedForward<? extends String, ? extends String>> expected = asList(
+                new CapturedForward<>(new Record<>("[alpha@100/200]", "2", 1_000L)),
+                new CapturedForward<>(new Record<>("[beta@100/200]", "1", 1_000L)),
+                new CapturedForward<>(new Record<>("[delta@200/300]", "1", 1_000L)),
+                new CapturedForward<>(new Record<>("[gamma@100/200]", "1", 1_000L)),
+                new CapturedForward<>(new Record<>("[gamma@200/300]", "1", 1_000L))
+            );
+
+            assertThat(capturedForwards, is(expected));
         } finally {
             Utils.delete(stateDir);
         }
     }
-
-    @SuppressWarnings("deprecation") // TODO will be fixed in KAFKA-10437
-    @Test
-    public void shouldFailWithLogging() {
```

Comment:
Can we remove those two tests?

Reply:
Yes, they are superceded by the new `MockProcessorContextStateStoreTest`, which verifies that any state store configured with logging or caching is rejected.